core/slee/services/sbbs/src/main/java/org/mobicents/gmlc/slee/MobileCoreNetworkInterfaceSbb.java

```diff
@@ -577,10 +577,10 @@ private void handleLocationResponse(MLPResponse.MLPResultType mlpResultType, CGI
 					getResponse.append(",vlrNumber=");
 					getResponse.append(response.vlr);
 
-                    this.sendHTTPResult(getResponse.toString());
+                    this.sendHTTPResult(HttpServletResponse.SC_OK, getResponse.toString());
                 }
                 else {
-                    this.sendHTTPResult(mlpClientErrorMessage);
+                    this.sendHTTPResult(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, mlpClientErrorMessage);
```

Comment:
500 error is quite generic, a per app error code translation should be implemented using proper http codes...,still 500 is better than nothing